core/src/main/java/org/apache/calcite/sql/SqlColumnAttributeTitle.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql;
+
+import org.apache.calcite.sql.parser.SqlParserPos;
+
+/**
+ * A {@code SqlColumnAttributeTitle} is the TITLE column attribute for a data
+ * type.
+ */
+public class SqlColumnAttributeTitle extends SqlColumnAttribute {
+
+  public final SqlNode titleString;
+
+  /**
+   * Creates a {@code SqlColumnAttributeTitle}.
+   *
+   * @param pos  Parser position, must not be null
+   * @param titleString String after the TITLE keyword
+   */
+  public SqlColumnAttributeTitle(SqlParserPos pos, SqlNode titleString) {
+    super(pos);
+    this.titleString = titleString;
+  }
+
+  @Override public void unparse(final SqlWriter writer, final int leftPrec,
+      final int rightPrec) {
+    writer.keyword("TITLE");
+    titleString.unparse(writer, 0, 0);
```

Comment:
May want to add a null check for titleString either here or in the constructor.

Reply:
Sure I could add a Objects.requireNonNull call in the constructor